indigo/bulk_creator.py

```diff
@@ -352,6 +404,18 @@ def create_work(self, row, idx):
                 else:
                     row.errors = str(e)
 
+    def create_work(self, row, idx):
+        # handle spreadsheet that still only uses 'principal'
+        row['stub'] = row.get('stub') if 'stub' in row else not row.get('principal')
+        row = self.validate_row(row)
```

Comment:
`validate_row` takes raw data and gives you something structured back. It's therefore responsible for giving you back something useful. It should also take care of `status` and `row_number`, otherwise you're splitting responsibility for the pristine state of `row` between two different methods.